dist/common/scripts/scylla_util.py

```diff
@@ -106,6 +110,137 @@ def curl(url, byte=False):
+        if dev[0:4] != "/dev":
+            dev = "/dev/%s" % dev
+        return os.path.exists(dev)
+
+    def get_file_size(self, filename):
+      "Get the file size by seeking at end"
+      fd = os.open(filename, os.O_RDONLY)
+      try:
+          return os.lseek(fd, 0, os.SEEK_END)
+      finally:
+          os.close(fd)
+
+    def _non_root_nvmes(self):
+        nvme_re = re.compile(r"nvme\d+n\d+$")
+
+        root_dev_candidates = [x for x in psutil.disk_partitions() if x.mountpoint == "/"]
+        if len(root_dev_candidates) != 1:
+            raise Exception("found more than one disk mounted at root'".format(root_dev_candidates))
+
+        root_dev = root_dev_candidates[0].device
+        nvmes_present = list(filter(nvme_re.match, os.listdir("/dev")))
+        return {"root": [root_dev], "ephemeral": [x for x in nvmes_present if not root_dev.startswith(os.path.join("/dev/", x))]}
+
+    def ephemeral_disks(self):
+        """Returns all ephemeral disks. Include standard SSDs and NVMe"""
+        return set(self._disks["ephemeral"])
+
+    def __populate_disks(self):
+        self._disks = {}
+        nvmes_present = self._non_root_nvmes()
+        for k, v in nvmes_present.items():
+            self._disks[k] = v
+
+    def __mac_address(self, nic='eth0'):
+        with open('/sys/class/net/{}/address'.format(nic)) as f:
+            return f.read().strip()
+
+    ## Use this to initialize basic data about the instance
+    def populate(self):
+        self._type = self.__instance_metadata("machine-type").split("/")[-1]
+        self._cpu = self.__cpu_count()
+        self._memory = self.__memory_size_gb()
+        self.__populate_disks()
+        self._populated = True
+
+    def instance(self):
+        """Returns which instance we are running in. i.e.: n2-standard-2"""
+        return self._type
+
+    def instance_size(self):
+        """Returns the size of the instance we are running in. i.e.: 2"""
+        return self._type.split("-")[2]
+
+    def instance_type(self):
+        """Returns the size of the instance we are running in. i.e.: standard"""
+        return self._type.split("-")[1]
+
+    def instance_class(self):
+        """Returns the class of the instance we are running in. i.e.: n1"""
+        return self._type.split("-")[0]
+
+    def is_supported_instance_class(self):
+        if self.instance_class() in ['n1', 'n2', 'm1', 'm2', 'c2']:
+            return True
+        return False
+
+    def is_supported_instance_size(self):  # at least 2 cpus are needed
+        if int(self.instance_size()) > 1:
+            return True
+        return False
+
+    def is_supported_instance_type(self):
+        if self.instance_type() in ['standard', 'highmem', 'ultramem', 'megamem']:
+            return True
+        return False
+
+    def is_supported_instance(self):
+        if self.is_supported_instance_class() and self.is_supported_instance_type() and self.is_supported_instance_size() and self._populated:
```

Comment:
Why do we want to limit the support to only these instances?
I think we should support any instance.